### PR TITLE
Add flag for selectively disabling choices in coverage report

### PR DESCRIPTION
**Pull request by dylant-da**
Addresses https://github.com/digital-asset/daml/issues/17409
Still needs docs - a summary of the workflows that are currently tested in compiler/damlc/tests/src/DamlcTest.hs would suffice I think.

### CODE REVIEW

File: compiler/damlc/lib/DA/Cli/Damlc/Test/TestResults.hs [hunk 1]
```diff
@@ -546,21 +554,32 @@ printTestCoverage showCoverage testResults@TestResults { templates, interfaceIns
...
+skipMatchingChoices namePred TestResults{..} =
+  TestResults
+    { templates = M.mapWithKey (S.filter . shouldntSkip) templates
+    , interfaces = M.mapWithKey (S.filter . shouldntSkip . unInterfaceIdentifier) interfaces
+    , interfaceInstances
+    , created
+    , exercised = M.mapWithKey (\choiceName -> M.filterWithKey (\tid _ -> shouldntSkip tid choiceName)) exercised
+    }
+  where
+    shouldntSkip :: TemplateIdentifier -> T.Text -> Bool
```
Comment (paulbrauner-da) [inline compiler/damlc/lib/DA/Cli/Damlc/Test/TestResults.hs:584]: Suggestion: rename to `shouldKeep` or `shouldStay` to avoid the double negation.

**Review by paulbrauner-da — APPROVED** [review]
Thank you for putting this together quickly!

**Review by akrmn — APPROVED** [review]
Looks great, thank you @dylant-da!

File: compiler/damlc/lib/DA/Cli/Damlc/Test/TestResults.hs (hunk 1 above)
Comment (dylant-da) [reply compiler/damlc/lib/DA/Cli/Damlc/Test/TestResults.hs]: Fixed :)

**Pull request merged**